libdrgn/type.c

```diff
@@ -987,6 +1021,54 @@ drgn_type_parameters_eq(struct drgn_type *a, struct drgn_type *b,
 	return NULL;
 }
 
+static struct drgn_error *drgn_type_template_parameters_eq(
+	struct drgn_type *a,
+	struct drgn_type *b,
+	struct drgn_type_pair_set *cache,
+	int *depth, bool *ret)
+{
+	struct drgn_template_parameter *templates_a, *templates_b;
+	size_t num_a, num_b, i;
+
+	num_a = drgn_type_num_template_parameters(a);
+	num_b = drgn_type_num_template_parameters(b);
+	if (num_a != num_b)
+		goto out_false;
+
+	templates_a = drgn_type_template_parameters(a);
+	templates_b = drgn_type_template_parameters(b);
+	for (i = 0; i < num_a; i++) {
+		struct drgn_qualified_type type_a, type_b;
+		struct drgn_error *err;
+
+		if (!templates_a[i].name && !templates_b[i].name)
+			continue;
```

Comment:
It turns out you copied a bug in `drgn_type_members_eq()`. I just fixed it in 4cbb9b552a161046bdb85b9814efc6c997484c32.